enhancements/cluster-logging/api-audit-log-policy.md

```diff
@@ -0,0 +1,254 @@
+- independent of the users desired policy and destinations.
+
+### Goals
+
+#### Fine grained control
+
+The [Kubernetes Policy][k8s-auditing] provides detailed tuning, and has been well tested by the Kubernetes community and Service Delivery.
+
+#### Multiple separate audit policies
+
+For managed clusters, the cluster manager and cluster owner may have different auditing needs.
+third party cluster auditing tools may introduce yet other auditing needs.
+
+To support _multiple_ independent audit log streams, we cannot configure filtering in the API server directly.
+Instead the API server will write complete logs (or some super-set of all desired logs)
+and the `ClusterLogForwarder` will apply a separate policy filter for each consumer.
+
+### Non-Goals
+
+- No help with semantics of audit events or design of audit policies.
+- No Hypershift support in this proposal -  [see separate hypershift proposal](hypershift-audit-logs.md)
+
+## Proposal
+
+### Workflow Description
+
+Cluster admin will
+1. Create a `Policy.audit.k8s.io` resource with the desired policy.
+2. Create an `audit` input in the `ClusterLogForwarder` linked to the policy.
+3. Create a `ClusterLogForwarder.pipeline` as normal to forward the resulting logs.
+
+### API Extensions
+
+#### New resource `Policy.audit.k8s.io`
+
+See [Kubernetes Documentation][k8s-auditing] for an example.
+
+#### Extensions to `ClusterLogForwarder.logging.openshift.io`
+
+Extended the `ClusterLogForwarder` input selector to include an `audit` section with a policy.
+For example:
+
+``` yaml
+  inputs:
+    - name: myAuditLogs
+      audit:
+        api:
+          myAuditPolicy # Name of a Policy.audit.k8s.io object in same namespace.
+  outputs:
+    - name: myAuditLogSore
+      type: any supported log output ...
+  pipelines:
+    - inputRefs: [myAuditLogs]
+      outputRefs: [myAuditLogStore]
+```
+
+FIXME UPDATE FROM DESIGN
+
+### Implementation Details
+
+There is already a go executable that does everything we need: 
+https://gitlab.cee.redhat.com/service/splunk-audit-exporter
+The "splunk" part is a misnomer, this is a filter from file/stdin/web-hook to stdout.
+
+This proposal only uses the audit log filtering features of the exporter.
+The exporter also has rate-limiting and other features that may be incorporated in future.
+
+For an input with an `apiServerPolicyRef`, the collector will
+- mount or copy the policy to the local file-system.
+- configure vector with an Exec source that runs the filter program as a separate process using the policy file.
+- vector reads stdout of the filter from the Exec source.
+- from there we route the events as normal
+
+Note: all of the above should support multiple distinct policies, so unique names for policy files etc.
+
+### Risks and Mitigations
+
+#### STS Authentication
+
+CLO needs to refresh tokens from the HCP in order to communicate w/ CloudWatch.
+Transplant the relevant code from the modified splunk-exporter used for Hypershift GA.
```

Comment:
Move this section to hypershift-audit-logs.md.  Any considerations for STS for non-hypershift ROSA?

Reply:
None that I know of, we already have STS support there. Again got my wires crossed.